File: Jack.Business/Status.vb
Public Class Status
    Implements ICrud(Of Model.Status, Integer)

    Public Sub New()

    End Sub

    Public Function Delete(oTipo As Model.Status) As Boolean Implements ICrud(Of Model.Status, Integer).Delete
        Dim oDados As Data.Status = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Status
            blnRetorno = oDados.Delete(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno

    End Function

    Public Function Find(Identifier As Integer) As Model.Status Implements ICrud(Of Model.Status, Integer).Find

        Dim oDados As Data.Status = Nothing
        Dim oRetorno As Model.Status = Nothing

        Try
            oDados = New Data.Status
            oRetorno = oDados.Find(Identifier)
        Catch ex As Exception
            oRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return oRetorno

    End Function

    Public Function Insert(oTipo As Model.Status) As Boolean Implements ICrud(Of Model.Status, Integer).Insert
        Dim oDados As Data.Status = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Status
            blnRetorno = oDados.Insert(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function

    Private Function LoadAll() As IList(Of Model.Status) Implements ICrud(Of Model.Status, Integer).LoadAll
        Dim oDados As Data.Status = Nothing
        Dim lstRetorno As IList(Of Model.Status) = Nothing

        Try
            oDados = New Data.Status
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function LoadForCriancas() As IList(Of Model.Status)
        Dim lstRetorno As IList(Of Model.Status) = Nothing

        Try
            lstRetorno = (From oDado As Model.Status In LoadAll() Where oDado.NivelStatus = "T" Or oDado.NivelStatus = "C" Select oDado).ToList()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        End Try

        Return lstRetorno
    End Function

    Public Function LoadForFamilia() As IList(Of Model.Status)
        Dim lstRetorno As IList(Of Model.Status) = Nothing

        Try
            lstRetorno = (From oDado As Model.Status In LoadAll() Where oDado.NivelStatus = "T" Or oDado.NivelStatus = "F" Select oDado).ToList()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        End Try

        Return lstRetorno
    End Function

    Public Function Update(oTipo As Model.Status) As Boolean Implements ICrud(Of Model.Status, Integer).Update
        Dim oDados As Data.Status = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Status
            blnRetorno = oDados.Update(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function
End Class

File: Jack.Business/Status.cs
using Jack.Model.DTOs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Jack.Business
{
    public class Status : ICrud<Model.Status, int>
    {


        public Status()
        {

        }

        public bool Delete(Model.Status oTipo)
        {
            Data.Status oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Status();
                blnRetorno = oDados.Delete(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;

        }

        public Model.Status Find(int Identifier)
        {

            Data.Status oDados = null;
            Model.Status oRetorno = null;

            try
            {
                oDados = new Data.Status();
                oRetorno = oDados.Find(Identifier);
            }
            catch (Exception ex)
            {
                oRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return oRetorno;

        }

        public bool Insert(Model.Status oTipo)
        {
            Data.Status oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Status();
                blnRetorno = oDados.Insert(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }

        public IList<Model.Status> LoadAll()
        {
            Data.Status oDados = null;
            IList<Model.Status> lstRetorno = null;

            try
            {
                oDados = new Data.Status();
                lstRetorno = oDados.LoadAll();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        public IList<DTOStatus> LoadForCriancas()
        {
            IList<DTOStatus> lstRetorno = null;

            try
            {
                lstRetorno = Load().Where(x => x.NivelStatus == "T" || x.NivelStatus == "C").ToList();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }

            return lstRetorno;
        }

        public IList<DTOStatus> LoadForFamilia()
        {
            IList<DTOStatus> lstRetorno = null;

            try
            {
                lstRetorno = Load().Where(x => x.NivelStatus == "T" || x.NivelStatus == "F").ToList();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }

            return lstRetorno;
        }

        public IList<DTOStatus> Load()
        {
            Data.Status oDados = null;
            IList<DTOStatus> lstRetorno = null;

            try
            {
                oDados = new Data.Status();
                lstRetorno = oDados.Load();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;
        }

        public bool Update(Model.Status oTipo)
        {
            Data.Status oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Status();
                blnRetorno = oDados.Update(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }
    }
}
